src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -140,7 +220,38 @@ struct OMapInnerNode
     buffer.replay(*this);
   }
 
-  internal_iterator_t get_containing_child(const std::string &key);
+  internal_const_iterator_t get_containing_child(const std::string &key);
+  internal_iterator_t lower_bound(const std::string &key) {
+    return string_lower_bound(key);
+  }
+
+  internal_iterator_t upper_bound(const std::string &key) {
+    return string_upper_bound(key);
+  }
+
+  bool is_in_range(const std::string &key) const {
+    return get_begin() <= key && get_end() > key;
+  }
+
+  ~OMapInnerNode() {
+    if (this->is_valid() && !this->is_pending()) {
```

Comment:
I understand this is consistent to the original implementations, can you help to elaborate why `!is_pending()` is required to call `destory()` ?

IIUC if this node is initial_pending, it should have a pending parent pointing to this node, right ?

Reply:
> I understand this is consistent to the original implementations, can you help to elaborate why `!is_pending()` is required to call `destory()` ?
> 
> IIUC if this node is initial_pending, it should have a pending parent pointing to this node, right ? Why need to skip `destroy()` in this case ?

It seems that a valid and pending extent is destroyed only when the transaction is invalidated and reset/destroyed, in which case the order of extents' destruction is not guaranteed and `destroy()` is not needed since the parents will also be removed from memory. Am I right?

Reply:
Can we add the following comments to `class parent_tracker_t` ?

IIUC:
```c++
// The sharable linker from the ChildNodes to the same ParentNode
//
// The indirection of child.parent_tracker.parent is necessary
// because otherwise we'll need to update every child's parent
// upon commiting a mutated extent.
```

Reply:
> It seems that a valid and pending extent is destroyed only when the transaction is invalidated and reset/destroyed, in which case the order of extents' destruction is not guaranteed and destroy() is not needed since the parents will also be removed from memory. Am I right?

I think, the order of destruction is unimportant (irrelavent) here because children helds their parent, which means the destructors of children are guaranteed to be called before the parent's.

Can we add the following comments to `destroy()` ?

IIUC:
```c++
// destroy() is allowed to be skipped for the pending extents,
// because they will be destroyed altogether upon transaction invalidation.
```

Reply:
Yeah, will do